src/components/TableauConnectorForm.js

```diff
@@ -51,7 +51,7 @@ class TableauConnectorForm extends Component {
       query: this.props.query || '',
       queryType: this.props.queryType || 'sql',
       showDatasetSelector: false,
-      writingQuery: !!this.props.query
+      writingQuery: !!this.props.query || !!this.props.addQuery
```

Comment:
what does `writingQuery` indicate? that we should expect a query to be in the query box?